crates/holochain_sqlite/src/db/access.rs

```diff
@@ -122,9 +122,14 @@ impl<Kind: DbKindT> DbRead<Kind> {
             .checkout_connection(self.read_semaphore.clone())
             .await?;
 
-        tokio::task::spawn_blocking(move || conn.execute_in_read_txn(f))
-            .await
-            .map_err(DatabaseError::from)?
+            // Once sync code starts in the spawn_blocking it cannot be cancelled BUT if we've run out of threads to execute blocking work on then
+            // this timeout should prevent the caller being blocked by this await that may not finish.
+            tokio::time::timeout(std::time::Duration::from_millis(ACQUIRE_TIMEOUT_MS.load(Ordering::Acquire)), tokio::task::spawn_blocking(move || {
```

Comment:
Described by the comment but just to make sure it's clear why I'm doing this. When Holochain runs out of threads to run blocking work on, it just locks up. The caller might free up resources by finishing rather than blocking and waiting for a thread so I've added this. It does work but obviously if something else is holding threads and not releasing them then Holochain is already in trouble so this won't solve it.